ocs_ci/utility/aws.py

```diff
@@ -301,11 +304,73 @@ def start_ec2_instance(self, instance_id, wait=False):
         """
         res = self.ec2_client.start_instances(InstanceIds=[instance_id])
         if wait:
-            for sample in utils.TimeoutSampler(
-                TIMEOUT, SLEEP, self.get_instances_status_by_id, instance_id
-            ):
-                if sample == RUNNING:
-                    return True
-            return False
+            instance = self.get_ec2_instance(instance_id)
+            instance.wait_until_running()
         state = res.get('StartingInstances')[0].get('CurrentState').get('Code')
-        return state == PENDING
+        return state == constants.INSTANCE_PENDING
+
+
+def get_instances_ids_and_names(instances):
+    """
+    Get the instances IDs and names according to nodes dictionary
+
+    Args:
+        instances (list): Nodes dictionaries, returned by 'oc get node -o yaml'
+
+    Returns:
+        tuple: lists of node IDs and names
+
+    """
+    instance_names = [node.get('metadata').get('name') for node in instances]
+    instance_ids = [
+        'i-' + node.get('spec').get(
+            'providerID'
+        ).partition('i-')[-1] for node in instances
+    ]
+    return instance_ids, instance_names
```

Comment:
Would it be better to just return a dict here like `{instance_id: instance_name}` instead of having to zip it up after the fact or is there a reason to keep them in separate lists?

Reply:
right, thanks. converted to dict